ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/avdtestbase.py

```diff
@@ -43,6 +43,162 @@ def render(self) -> dict:
+        LOGGER.log(log_level, log_msg)
+
+    def is_peer_available(self, peer: str) -> bool:
+        """
+        Check if a peer is deployed by looking at his `is_deployed` key.
+
+        Args:
+            peer (str): The peer to verify.
+
+        Returns:
+            bool: True if the peer is deployed, False otherwise.
+        """
+        if peer not in self.hostvars:
+            self.log_message(message=f"Peer '{peer}' is not configured by AVD.")
+            return False
+        elif not get(self.hostvars[peer], "is_deployed", default=True):
+            self.log_message(message=f"Peer '{peer}' is marked as not deployed.")
+            return False
+        return True
+
+    def get_interface_ip(self, interface_model: str, interface_name: str, host: str | None = None) -> str | None:
+        """
+        Retrieve the IP address of a specified host interface.
+
+        Args:
+            interface_model (str): Interface model in the structured config (e.g., ethernet_interfaces).
+            interface_name (str): Interface name to retrive the IP.
+            host (str): Host to verify. Defaults to the host running the test.
+
+        Returns:
+            str | None: IP address of the host interface or None if unavailable.
+        """
+        host = host or self.device_name
+
+        try:
+            if host not in self.hostvars:
+                raise AristaAvdError(f"Host '{host}' is missing from the hostvars.")
+            peer_interfaces = get(self.hostvars[host], interface_model, required=True)
+            peer_interface = get_item(peer_interfaces, "name", interface_name, required=True)
+            return get(peer_interface, "ip_address", required=True)
+        except AristaAvdMissingVariableError:
+            self.log_message(message=f"Host '{host}' interface '{interface_name}' IP address is unavailable.", logging_level="WARNING")
+            return None
+
+    def logged_get(self, key: str, host: str | None = None, logging_level: str = "WARNING"):
+        """
+        Attempts to retrieve a value associated with a given key from structured_config and logs if it's missing.
+
+        Args:
+            key (str): The key to retrieve.
+            host (str | None): The host from which to retrieve the key. Defaults to the device running the test.
+            logging_level (str): The logging level to use for the log message.
+        """
+        host = host or self.device_name
+
+        try:
+            data = get(self.hostvars[host], key, required=True)
+            return data
+        except AristaAvdMissingVariableError:
+            self.log_message(key=key, logging_level=logging_level)
+            return None
+
+    def validate_data(
+        self,
+        data: dict | None = None,
+        data_path: str | None = None,
+        host: str | None = None,
+        required_keys: str | list[str] | None = None,
+        Logging_level: str | None = None,
```

Comment:
don't use capitalized names for arguments
```suggestion
        logging_level: str | None = None,
```

Reply:
Good catch. Find/replace messing up everything

Reply:
Fixed in 7c1821824bc682cfd9015747f8841e2e8f84991f